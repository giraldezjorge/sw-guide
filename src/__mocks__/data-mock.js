
export const mockCharacter = {
  name: 'Luke Skywalker',
  height: '172',
  mass: '77',
  hair_color: 'blond',
  skin_color: 'fair',
  eye_color: 'blue',
  birth_year: '19BBY',
  gender: 'male',
  url: 'test/people/1/'
}

export const mockCharacters = [
  mockCharacter,
  {
    name: 'C-3PO',
    height: '167',
    mass: '75',
    hair_color: 'n/a',
    skin_color: 'gold',
    eye_color: 'yellow',
    birth_year: '112BBY',
    gender: 'n/a',
    url: 'test/people/2/'
  },
  {
    name: 'R2-D2',
    height: '96',
    mass: '32',
    hair_color: 'n/a',
    skin_color: 'white, blue',
    eye_color: 'red',
    birth_year: '33BBY',
    gender: 'n/a',
    url: 'test/people/3/'
  }
]

export const mockFilm = {
  title: 'A New Hope',
  episode_id: 4,
  opening_crawl: 'It is a period of civil war.\r\nRebel spaceships, striking\r\nfrom a hidden base, have won\r\ntheir first victory against\r\nthe evil Galactic Empire.\r\n\r\nDuring the battle, Rebel\r\nspies managed to steal secret\r\nplans to the Empire\'s\r\nultimate weapon, the DEATH\r\nSTAR, an armored space\r\nstation with enough power\r\nto destroy an entire planet.\r\n\r\nPursued by the Empire\'s\r\nsinister agents, Princess\r\nLeia races home aboard her\r\nstarship, custodian of the\r\nstolen plans that can save her\r\npeople and restore\r\nfreedom to the galaxy....',
  director: 'George Lucas',
  producer: 'Gary Kurtz, Rick McCallum',
  release_date: '1977-05-25',
  url: 'test/films/1/',
  characters: [
    'test/people/1/'
  ]
}

export const mockFilms = [
  mockFilm,
  {
    title: 'The Empire Strikes Back',
    episode_id: 5,
    opening_crawl: 'It is a dark time for the\r\nRebellion. Although the Death\r\nStar has been destroyed,\r\nImperial troops have driven the\r\nRebel forces from their hidden\r\nbase and pursued them across\r\nthe galaxy.\r\n\r\nEvading the dreaded Imperial\r\nStarfleet, a group of freedom\r\nfighters led by Luke Skywalker\r\nhas established a new secret\r\nbase on the remote ice world\r\nof Hoth.\r\n\r\nThe evil lord Darth Vader,\r\nobsessed with finding young\r\nSkywalker, has dispatched\r\nthousands of remote probes into\r\nthe far reaches of space....',
    director: 'Irvin Kershner',
    producer: 'Gary Kurtz, Rick McCallum',
    release_date: '1980-05-17',
    url: 'test/films/2/',
    characters: [
      'test/people/1/'
    ]
  }
]
