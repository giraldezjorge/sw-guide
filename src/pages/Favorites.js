import React from 'react'
import { connect } from 'react-redux'
import { fetchFavorites } from '../actions'
import FilmsGrid from '../components/films-grid'
import CharactersGrid from '../components/characters-list'

class Favorites extends React.Component {
  componentDidMount () {
    this.props.fetchFavorites()
  }

  render () {
    const { films, characters } = this.props
    return (
      <section className="section">
        <div className="columns is-variable is-5">
          <div className="column">
            <FilmsGrid films={films} title={'Favorite films'} emptyMsg={'There are no favorite films'} />
          </div>
          <div className="column">
            <CharactersGrid characters={characters} title={'Favorite characters'} emptyMsg={'There are no favorite characters'} />
          </div>
        </div>
      </section>
    )
  }
}

const mapStateToProps = (state) => {
  const { favorites } = state
  return {
    films: favorites.filter(fav => fav.episode_id !== undefined) || [],
    characters: favorites.filter(fav => fav.gender !== undefined) || []
  }
}

export default connect(
  mapStateToProps,
  { fetchFavorites }
)(Favorites)
