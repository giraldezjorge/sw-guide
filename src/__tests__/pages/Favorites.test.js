import React from 'react'
import { mockFilms, mockCharacter } from '../../__mocks__/data-mock'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import Favorites from '../../pages/Favorites'

const mockStore = configureStore([])

describe('Favorites page', () => {
  it('shows empty favorite lists', () => {
    const store = mockStore({ favorites: [] })

    const component = renderer.create(
      <Provider store={store}>
        <Favorites />
      </Provider>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  }),

  it('shows empty film favorite list and character list', () => {
    const store = mockStore({ favorites: [mockCharacter] })

    const component = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <Favorites />
        </Provider>
      </MemoryRouter>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  }),

  it('shows film favorite list and empty character list', () => {
    const store = mockStore({ favorites: mockFilms })

    const component = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <Favorites />
        </Provider>
      </MemoryRouter>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  }),

  it('shows film favorite list and character list', () => {
    const store = mockStore({ favorites: [...mockFilms, mockCharacter] })

    const component = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <Favorites />
        </Provider>
      </MemoryRouter>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
