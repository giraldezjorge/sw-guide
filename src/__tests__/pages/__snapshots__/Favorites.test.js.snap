// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites page shows empty favorite lists 1`] = `
<section
  className="section"
>
  <div
    className="columns is-variable is-5"
  >
    <div
      className="column"
    >
      <div
        className="container"
      >
        <p
          className="title is-3"
        >
          Favorite films
        </p>
        <div
          className="columns is-multiline"
        >
          <div
            className="column"
          >
            <p
              className="empty-msg"
            >
              There are no favorite films
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="column"
    >
      <div
        className="container"
      >
        <p
          className="title is-3"
        >
          Favorite characters
        </p>
        <div
          className="columns is-multiline"
        />
        <div
          className="column"
        >
          <p
            className="empty-msg"
          >
            There are no favorite characters
          </p>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`Favorites page shows empty film favorite list and character list 1`] = `
<section
  className="section"
>
  <div
    className="columns is-variable is-5"
  >
    <div
      className="column"
    >
      <div
        className="container"
      >
        <p
          className="title is-3"
        >
          Favorite films
        </p>
        <div
          className="columns is-multiline"
        >
          <div
            className="column"
          >
            <p
              className="empty-msg"
            >
              There are no favorite films
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="column"
    >
      <div
        className="container"
      >
        <p
          className="title is-3"
        >
          Favorite characters
        </p>
        <div
          className="columns is-multiline"
        >
          <div
            className="character column is-half"
          >
            <div
              className="sc-bwzfXH kZUMPF card character-item"
              onClick={[Function]}
            >
              <div
                className="card-content"
              >
                <div
                  className="media"
                >
                  <div
                    className="media-left"
                  >
                    <figure
                      className="image is-24x24"
                    >
                      <img
                        alt="Character"
                        src="character-icon.png"
                      />
                    </figure>
                  </div>
                  <div
                    className="media-content"
                  >
                    <p
                      className="title is-6"
                    >
                      Luke Skywalker
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`Favorites page shows film favorite list and character list 1`] = `
<section
  className="section"
>
  <div
    className="columns is-variable is-5"
  >
    <div
      className="column"
    >
      <div
        className="container"
      >
        <p
          className="title is-3"
        >
          Favorite films
        </p>
        <div
          className="columns is-multiline"
        >
          <div
            className="film column is-one-quarter"
          >
            <div
              className="sc-bdVaJa fVCJpx card"
              onClick={[Function]}
            >
              <div
                className="card-image"
              >
                <figure
                  className="image is-4by3"
                >
                  <img
                    alt="Thumbnail"
                    src="sw_5.jpg"
                  />
                </figure>
              </div>
              <div
                className="card-content"
              >
                <div
                  className="media"
                >
                  <div
                    className="media-content"
                  >
                    <p
                      className="subtitle is-6"
                    >
                      Episode#4
                    </p>
                    <p
                      className="title is-4"
                    >
                      A New Hope
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="film column is-one-quarter"
          >
            <div
              className="sc-bdVaJa fVCJpx card"
              onClick={[Function]}
            >
              <div
                className="card-image"
              >
                <figure
                  className="image is-4by3"
                >
                  <img
                    alt="Thumbnail"
                    src="sw_6.jpg"
                  />
                </figure>
              </div>
              <div
                className="card-content"
              >
                <div
                  className="media"
                >
                  <div
                    className="media-content"
                  >
                    <p
                      className="subtitle is-6"
                    >
                      Episode#5
                    </p>
                    <p
                      className="title is-4"
                    >
                      The Empire Strikes Back
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="column"
    >
      <div
        className="container"
      >
        <p
          className="title is-3"
        >
          Favorite characters
        </p>
        <div
          className="columns is-multiline"
        >
          <div
            className="character column is-half"
          >
            <div
              className="sc-bwzfXH kZUMPF card character-item"
              onClick={[Function]}
            >
              <div
                className="card-content"
              >
                <div
                  className="media"
                >
                  <div
                    className="media-left"
                  >
                    <figure
                      className="image is-24x24"
                    >
                      <img
                        alt="Character"
                        src="character-icon.png"
                      />
                    </figure>
                  </div>
                  <div
                    className="media-content"
                  >
                    <p
                      className="title is-6"
                    >
                      Luke Skywalker
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`Favorites page shows film favorite list and empty character list 1`] = `
<section
  className="section"
>
  <div
    className="columns is-variable is-5"
  >
    <div
      className="column"
    >
      <div
        className="container"
      >
        <p
          className="title is-3"
        >
          Favorite films
        </p>
        <div
          className="columns is-multiline"
        >
          <div
            className="film column is-one-quarter"
          >
            <div
              className="sc-bdVaJa fVCJpx card"
              onClick={[Function]}
            >
              <div
                className="card-image"
              >
                <figure
                  className="image is-4by3"
                >
                  <img
                    alt="Thumbnail"
                    src="sw_5.jpg"
                  />
                </figure>
              </div>
              <div
                className="card-content"
              >
                <div
                  className="media"
                >
                  <div
                    className="media-content"
                  >
                    <p
                      className="subtitle is-6"
                    >
                      Episode#4
                    </p>
                    <p
                      className="title is-4"
                    >
                      A New Hope
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="film column is-one-quarter"
          >
            <div
              className="sc-bdVaJa fVCJpx card"
              onClick={[Function]}
            >
              <div
                className="card-image"
              >
                <figure
                  className="image is-4by3"
                >
                  <img
                    alt="Thumbnail"
                    src="sw_6.jpg"
                  />
                </figure>
              </div>
              <div
                className="card-content"
              >
                <div
                  className="media"
                >
                  <div
                    className="media-content"
                  >
                    <p
                      className="subtitle is-6"
                    >
                      Episode#5
                    </p>
                    <p
                      className="title is-4"
                    >
                      The Empire Strikes Back
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="column"
    >
      <div
        className="container"
      >
        <p
          className="title is-3"
        >
          Favorite characters
        </p>
        <div
          className="columns is-multiline"
        />
        <div
          className="column"
        >
          <p
            className="empty-msg"
          >
            There are no favorite characters
          </p>
        </div>
      </div>
    </div>
  </div>
</section>
`;
